gateway/gateway.go

```diff
@@ -252,69 +235,102 @@ func (g Gateway) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	if idx != -1 {
 		contentType = contentType[:idx]
 	}
-	codec := g.codecs[strings.ToLower(contentType)]
-	if codec == nil {
+	var codec rpc.Codec
+	// if no contentType was sent, assume the first codec if only one in list
+	// see: https://github.com/gorilla/rpc/pull/42/
+	if contentType == "" && len(g.codecs) == 1 {
+		for _, c := range g.codecs {
```

Comment:
oooh ok. should this loop break immediately then? I guess it doesn't matter. Maybe throw a small comment there since that just looks odd